internal/k3s/upgrade.go

```diff
@@ -0,0 +1,66 @@
+package k3s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/weka/gohomecli/internal/bundle"
+	"golang.org/x/mod/semver"
+)
+
+var ErrNotExist = errors.New("k3s not exists")
+
+type UpgradeConfig struct {
+	BundlePath string
+}
+
+func Upgrade(ctx context.Context, c UpgradeConfig) error {
+	if !hasK3S() {
+		return ErrNotExist
+	}
+
+	file, version, err := findBundle(c.BundlePath)
+	if err != nil {
+		return err
+	}
+
+	curVersion, err := getK3SVersion(k3sBinary)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Found k3s bundle %q, current version %q\n", version, curVersion)
+
+	if semver.Compare(version, curVersion) < 0 {
+		fmt.Println("Downgrading kubernetes cluster is not possible")
+		return nil
+	}
+
+	fmt.Println("Starting K3S upgrade...")
```

Comment:
we have zerolog logging configured, you may use logger instead

Reply:
agree, logging wasn't my main priority at that time